src/pulse3D/plate_recording.py

```diff
@@ -557,15 +557,18 @@ def to_dataframe(self) -> pd.DataFrame:
 
         data = {"Time (s)": pd.Series(interp_timepoints)}
 
-        output_stim_data = not self.is_optical_recording and first_well.version >= VersionInfo.parse("1.0.0")
-
-        # add stim timepoints
-        if output_stim_data:
-            aggregate_stim_timepoints_us = aggregate_timepoints(
-                [session_data[0] for wf in self for session_data in wf.stim_sessions]
+        if not ignore_stim_data:
```

Comment:
There is already conditional logic in here that handles this, it would be better to combine this new condition with the existing ones. If we just created a new if/else branch every time a new condition was added, it would quickly become a mess of if/else branching that makes the actual logic unclear. You can simplify this by just doing something like 
```
output_stim_data = include_stim_data and first_well.version >= VersionInfo.parse(
   MIN_FILE_VERSION_FOR_STIM_INTERPOLATION
)
```